From 45925668371f89ac0dfffe59afac77e87588fc1e Mon Sep 17 00:00:00 2001
From: Yang Dong <b56112@freescale.com>
Date: Thu, 20 Apr 2017 10:27:12 +0300
Subject: [PATCH 24/31] MGS-470 [#ccc] cannot trace mesa-demos/vertexrate

Root cause: glxGetCurrentDisplay goes into a infinite loop.

Solution: Before calling _glxGetCurrentDisplay, check the value of it.
If _glxGetCurrentDisplay reference glxGetCurrentDisplay, redirect it to proper lib.

Signed-off-by: Yang Dong <b56112@freescale.com>
---
 wrappers/glxtrace.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/wrappers/glxtrace.py b/wrappers/glxtrace.py
index d9ffe56..e2d0fc4 100644
--- a/wrappers/glxtrace.py
+++ b/wrappers/glxtrace.py
@@ -66,6 +66,11 @@ class GlxTracer(GlTracer):
         if function.name in self.destroyContextFunctionNames:
             print '    gltrace::releaseContext((uintptr_t)ctx);'
 
+        if function.name == 'glXGetCurrentDisplay':
+            print '    if(_glXGetCurrentDisplay == &glXGetCurrentDisplay ){'
+            print '        _glXGetCurrentDisplay = (PFN_GLXGETCURRENTDISPLAY)_getPublicProcAddress("glXGetCurrentDisplay");'
+            print '    }'
+
         GlTracer.traceFunctionImplBody(self, function)
 
         if function.name in self.createContextFunctionNames:
-- 
2.7.4

